src/psyclone/parse/module_manager.py

```diff
@@ -0,0 +1,324 @@
+            either the cached data nor in the search path.
+
+        '''
+        mod_lower = module_name.lower()
+
+        # First check if we already know about this file:
+        mod_info = self._mod_2_filename.get(mod_lower, None)
+        if mod_info:
+            return mod_info
+
+        # If not, check the search paths. To avoid frequent accesses to
+        # the directories, we search directories one at a time, and
+        # add the list of all files in that directory to our cache
+        # _mod_2_filename
+        while self._search_paths:
+            # Get the first element from the search path list:
+            directory = self._search_paths.pop(0)
+            self._add_all_files_from_dir(directory)
+            mod_info = self._mod_2_filename.get(mod_lower, None)
+            if mod_info:
+                return mod_info
+
+        raise FileNotFoundError(f"Could not find source file for module "
+                                f"'{module_name}'.")
+
+    # ------------------------------------------------------------------------
+    def get_modules_in_file(self, filename):
+        # pylint: disable=no-self-use
+        '''This function returns the list of modules defined in the specified
+        file. The base implementation uses the coding style: the file
+        `a_mod.f90` implements the module `a_mod`. This function can be
+        implemented in a derived class to actually parse the source file if
+        required.
+
+        :param str filename: the file name for which to find the list \
+            of modules it contains.
+
+        :returns: the list of all modules contained in the specified file.
+        :rtype: List[str]
+
+        '''
+        basename = os.path.basename(filename)
+        root, _ = os.path.splitext(basename)
+        if root[-4:].lower() == "_mod":
+            return [root]
+
+        return []
+
+    # ------------------------------------------------------------------------
+    def get_all_dependencies_recursively(self, all_mods):
+        '''This function collects recursively all module dependencies
+        for any of the modules in the ``all_mods`` set. I.e. it will
+        add all modules used by any module listed in ``all_mods``,
+        and any modules used by the just added modules etc. In the end,
+        it will return a dictionary that for each module lists which
+        module this module depends on. This dictionary will be complete,
+        i.e. all modules that are required for the original set of modules
+        (and that could be found) will be a key in the dictionary. It will
+        include the original set of modules as well.
+
+        If a module cannot be found (e.g. its path was not given to the
+        ModuleManager, or it might be a system module for which the sources
+        are not available, a message will be printed, and this module will
+        be ignored (i.e. not listed in any dependencies)
```

Comment:
In #1990 I've added a config option to list which modules to ignore without any messages. The patch also modifies the module_manager, so I don't think it will cleanly apply here, so I'd prefer to wait to do this as part of the next PR.

Reply:
OK.